File: android/sdk/ssdid-core/src/main/java/my/ssdid/sdk/domain/transport/dto/NotifyDto.kt
package my.ssdid.sdk.domain.transport.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

// ---------- Inbox ----------

@Serializable
data class NotifyDevice(
    val platform: String,
    val token: String
)

@Serializable
data class RegisterInboxRequest(
    val devices: List<NotifyDevice>
)

@Serializable
data class RegisterInboxResponse(
    @SerialName("inbox_id") val inboxId: String,
    @SerialName("inbox_secret") val inboxSecret: String
)

@Serializable
data class UpdateDevicesRequest(
    val devices: List<NotifyDevice>
)

// ---------- Mailbox ----------

@Serializable
data class CreateMailboxRequest(
    @SerialName("inbox_id") val inboxId: String,
    @SerialName("mailbox_id") val mailboxId: String
)

// ---------- Pending notifications ----------

@Serializable
data class PendingNotification(
    @SerialName("notification_id") val notificationId: String,
    @SerialName("mailbox_id") val mailboxId: String,
    val payload: String,
    val priority: String = "normal",
    @SerialName("received_at") val receivedAt: String? = null
)

@Serializable
data class PendingNotificationsResponse(
    val notifications: List<PendingNotification>
)

// ---------- Publish (service-side, included for completeness) ----------

@Serializable
data class PublishRequest(
    @SerialName("mailbox_id") val mailboxId: String,
    val payload: String,
    val priority: String = "normal"
)

// ---------- Service registration (service-side) ----------

@Serializable
data class RegisterServiceRequest(
    val name: String,
    @SerialName("public_key") val publicKey: String
)

@Serializable
data class RegisterServiceResponse(
    @SerialName("service_id") val serviceId: String,
    @SerialName("api_key") val apiKey: String
)

File: sdk/ios/SsdidCore/Sources/SsdidCore/Domain/Transport/DTO/NotifyDto.swift
import Foundation

/// Request/response DTOs for SSDID Notify service interactions.

// MARK: - Device

public struct NotifyDevice: Codable {
    public let platform: String
    public let token: String
}

// MARK: - Inbox

public struct InboxRegisterRequest: Codable {
    public let devices: [NotifyDevice]
}

public struct InboxRegisterResponse: Codable {
    public let inboxId: String
    public let inboxSecret: String

    enum CodingKeys: String, CodingKey {
        case inboxId = "inbox_id"
        case inboxSecret = "inbox_secret"
    }
}

public struct InboxUpdateDevicesRequest: Codable {
    public let devices: [NotifyDevice]
}

// MARK: - Mailbox

public struct MailboxCreateRequest: Codable {
    public let inboxId: String
    public let mailboxId: String

    enum CodingKeys: String, CodingKey {
        case inboxId = "inbox_id"
        case mailboxId = "mailbox_id"
    }
}

// MARK: - Pending Notifications

public struct NotifyNotification: Codable {
    public let notificationId: String
    public let mailboxId: String
    public let payload: String
    public let priority: String?
    public let receivedAt: String?

    enum CodingKeys: String, CodingKey {
        case notificationId = "notification_id"
        case mailboxId = "mailbox_id"
        case payload
        case priority
        case receivedAt = "received_at"
    }
}

public struct PendingNotificationsResponse: Codable {
    public let notifications: [NotifyNotification]
}
